src/shell/Models/ShellExtension.cs

```diff
@@ -0,0 +1,122 @@
+            if(string.IsNullOrWhiteSpace(subCommandName) || index < 0)
+            {
+                throw new Exception("Invalid subcommand.");
+            }
+            Process process = CreateNewProcess(args, input, index);
+            process.Start();
+            process.WaitForExit();
+            using var reader = new StreamReader(process.StandardOutput.BaseStream);
+            string output = await reader.ReadToEndAsync();
+            var subCommand = this.Find(subCommandName);
+            if (subCommand != null && subCommand.InvokeContract)
+            {
+                await HandleTransactionsAsync(input, chainManagerFactory, txExecutorFactory, output, subCommand).ConfigureAwait(false);
+            }
+            else
+            {
+                await sdoutWriter.WriteLineAsync(output);
+            }
+
+            var error = await process.StandardError.ReadToEndAsync();
+            if (!string.IsNullOrWhiteSpace(error))
+            {
+                await errorWriter.WriteLineAsync(error);
+            }
+            return process.ExitCode;
+        }
+
+        private async Task<TransactionExecutor> HandleTransactionsAsync(string input, ExpressChainManagerFactory? chainManagerFactory, TransactionExecutorFactory? txExecutorFactory, string output, SubCommand subCommand)
+        {
+            var invokeParams = JsonConvert.DeserializeObject<InvocationParameter>(output);
+            if (invokeParams == null)
+            {
+                throw new Exception("Invalid invocation parameters from the subcommand.");
+            }
+            if (chainManagerFactory == null || txExecutorFactory == null)
+            {
+                throw new Exception("ChainManagerFactory or TransactionExecutorFactory is not initialized.");
+            }
+
+            var (chainManager, _) = chainManagerFactory.LoadChain(input);
+            var txExec = txExecutorFactory.Create(chainManager, invokeParams.Trace, invokeParams.Json);
+            var script = new Script(Convert.FromBase64String(invokeParams.Script));
+            if (subCommand.Safe)
+            {
+                await txExec.InvokeForResultsAsync(script, invokeParams.Account ?? string.Empty, WitnessScope.CalledByEntry).ConfigureAwait(false);
+            }
+            else
+            {
+                var password = string.Empty;
+                if (!invokeParams.Account.IsWIFString())
+                {
+                    password = chainManager.Chain.ResolvePassword(invokeParams.Account, password);
+                    await txExec.ContractInvokeAsync(script, invokeParams.Account, password, WitnessScope.CalledByEntry, 0);
+                }
+            }
+
+            return txExec;
+        }
+
+        private Process CreateNewProcess(string[] args, string input, int index)
+        {
+            var process = new Process();
+            process.StartInfo.FileName = this.MapsToCommand;
+            var arguments = new List<string>();
```

Comment:
is this needed?

Reply:
removed 98:     var arguments = new List<string>();